awswrangler/distributed/ray/modin/_data_types.py

```diff
@@ -0,0 +1,30 @@
+"""Internal (private) Data Types Module."""
+
+from typing import Dict, List, Optional
+
+import modin.pandas as pd
+import pyarrow as pa
+import ray
+
+from awswrangler._data_types import pyarrow_types_from_pandas
+from awswrangler.distributed.ray._core import ray_get, ray_remote
+
+
+def pyarrow_types_from_pandas_distributed(
+    df: pd.DataFrame, index: bool, ignore_cols: Optional[List[str]] = None, index_left: bool = False
+) -> Dict[str, pa.DataType]:
+    """Extract the related Pyarrow data types from a pandas DataFrame."""
+    func = ray_remote(pyarrow_types_from_pandas)
+    block_object_refs = ray.data.from_modin(df).get_internal_block_refs()
+    result = ray_get(
+        [
+            func(
+                df=object_ref,
+                index=index,
+                ignore_cols=ignore_cols,
+                index_left=index_left,
+            )
+            for object_ref in block_object_refs
+        ]
+    )
+    return {k: v for d in result for k, v in d.items()}
```

Comment:
We could make use of [this function](https://arrow.apache.org/docs/python/generated/pyarrow.unify_schemas.html). If the data types are incompatible, it will just fail. But catching the exception will at least give us a chance to log the error along with details.

The log can then be something like:
> Data schemas across data blocks are incompatible. Sampling to one of the schemas